File: Praticas/C/botao.c
#include <wiringPi.h>

#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>



/*
   PINO 12 -> GPIO1	[Outras Linguagens eh GPIO18]

 */
int main() {
	/* Verifica se WiringPiSetup esta OK */
	if (wiringPiSetup () == -1)
		exit (1) ;

	/* Configura GPIO1 como INPUT */
	pinMode(1, INPUT);


	printf("\nIniciando Aplicacao C com WiringPI\n");

	/* LOOP Leitura Botao */
	while(1) {
		if( ! digitalRead(1) ) {
			printf("BOTAO GPIO[1] ACIONADO\n");
		}
		sleep(1);
	}

	return 0 ;
}

File: Praticas/Perl/botao.pl
#! /usr/bin/perl

use strict;
use warnings;

$SIG{INT}  = \&control_signal;
$SIG{TERM} = \&control_signal;


my $fh; 

# Exportando o GPIO18
open($fh, '>', '/sys/class/gpio/export');
print $fh "18";
close $fh;


# Setando a direcao do GPIO18 como INPUT
open($fh, '>', '/sys/class/gpio/gpio18/direction');
print $fh "in";
close $fh;


print "LENDO BOTAO GPIO18 COM PERL\n\n";



my $valor_botao;

# LOOP LEITURA BOTAO
while (1){
	open FDIN, "</sys/class/gpio/gpio18/value" or die $!;
	$valor_botao = <FDIN>;
	if($valor_botao == "0") {
		print "BOTAO ACIONADO\n";
	}
	close FDIN;
	sleep(1);
}



# EVENTO QUE CAPTURA SINGAL DE CTRL+C E ENCERRA APLICACAO
sub control_signal {
	print "Sinal recebido, aguarde...";

	open($fh, '>', '/sys/class/gpio/unexport');
	print $fh "18";
	close $fh;
	exit 0;

}
